components/debugbar/emailssent.tsx

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useState} from 'react';
+import {useSelector} from 'react-redux';
+import cn from 'classnames';
+import {FixedSizeList as List} from 'react-window';
+
+import {getEmailsSent} from 'mattermost-redux/selectors/entities/debugbar';
+
+import {GenericModal} from '@mattermost/components';
+import {DebugBarEmailSent} from '@mattermost/types/debugbar';
+
+import Time from './time';
+
+type Props = {
+    filter: string;
+    height: number;
+}
+
+type RowProps = {
+    data: DebugBarEmailSent[];
+    index: number;
+    style: any;
+}
+
+function EmailsSent({filter, height}: Props) {
+    const [viewEmail, setViewEmail] = useState<DebugBarEmailSent|null>(null);
+    let emails = useSelector(getEmailsSent);
+    if (filter !== '') {
+        emails = emails.filter((v) => JSON.stringify(v).indexOf(filter) !== -1);
```

Comment:
This works for now. Wanted to point out that some of the fields of the email will be identical across emails like SMTP config, and it will include html content so searching things like html tag names will return matches.

*If* we wanted to refine in the future we could do something like parsing the html document and grabbing the inner text so that only the text portion is searchable.

```tsx
const el = document.createElement( 'html' );
el.innerHTML = v.htmlBody;
contentToSearch = `${v.category} ${v.cc} ${v.inReplyTo} ${v.messageID} ${v.references} ${v.subject} ${v.time} ${v.to} ${v.type} ${el.innerText}`

```

And then again, maybe we *want* to be able to search for html class names that appear (or fail to appear)

Reply:
For now is going to be like this, after this is merged, we can improve this using something like https://lunrjs.com/ or (elastic lunr)